src/Plugin/Reverse/Reverse.cpp

```diff
@@ -0,0 +1,236 @@
+
+  Reverse.cpp - DPF + Zyn Plugin for Reverse
+  Copyright (C) 2015 Filipe Coelho
+  Author: Filipe Coelho
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+*/
+
+// DPF includes
+#include "../AbstractFX.hpp"
+
+// ZynAddSubFX includes
+#include "Effects/Reverse.h"
+
+/* ------------------------------------------------------------------------------------------------------------
+ * Reverse plugin class */
+
+class ReversePlugin : public AbstractPluginFX<zyn::Reverse>
+{
+public:
+    ReversePlugin()
+        : AbstractPluginFX(9, 5) {}
+
+    void setSpeedfactor(float factor)
+    {
+        effect->speedfactor=factor;
+    }
+
+protected:
+   /* --------------------------------------------------------------------------------------------------------
+    * Information */
+
+   /**
+      Get the plugin label.
+      This label is a short restricted name consisting of only _, a-z, A-Z and 0-9 characters.
+    */
+    const char* getLabel() const noexcept override
+    {
+        return "Reverse";
+    }
+
+   /**
+      Get an extensive comment/description about the plugin.
+    */
+    const char* getDescription() const noexcept override
+    {
+        return "Reverse is a host synchronized reverse delay effect with multiple trigger modes";
+    }
+
+   /**
+      Get the plugin unique Id.
+      This value is used by LADSPA, DSSI and VST plugin formats.
+    */
+    int64_t getUniqueId() const noexcept override
+    {
+        return d_cconst('Z', 'X', 'r', 's');
+    }
+
+    zyn::Reverse* instantiateFX(zyn::EffectParams pars) override
+    {
+        return new zyn::Reverse(pars, &time);
+    }
+
+
+    /**
+      Get the current value of a parameter.
+      The host may call this function from any context, including realtime processing.
+    */
+    float getParameterValue(uint32_t index) const override
+    {
+        switch(index)
+        {
+            case 4:
+                return AbstractPluginFX::getParameterValue(index)-2;
+                break;
+            case 5:
+                return numerator;
+                break;
+            case 6:
+                return denominator;
+                break;
+            default:
+                return AbstractPluginFX::getParameterValue(index);
+                break;
+         }
+
+    }
+
+   /**
+      Change a parameter value.
+      The host may call this function from any context, including realtime processing.
+      When a parameter is marked as automable, you must ensure no non-realtime operations are performed.
+      @note This function will only be called for parameter inputs.
+    */
+    void setParameterValue(uint32_t index, float value) override
+    {
+        if(index==4)
+        {
+            AbstractPluginFX::setParameterValue(index, value+2);
```

Comment:
Same as before, should the offset be applied to the index?

Reply:
hmm, maybe it's not a bug?

Reply:
it's a mapping of the 4 sync modes to the two supported in plugin mode
sync to noteOn/Off is not available in a plugin.